src/App.tsx

```diff
@@ -3,12 +3,33 @@ import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
-type State = {};
+type State = {
+  query: string;
+};
 
 export class App extends React.Component<{}, State> {
-  state: State = {};
+  state: State = {
+    query: '',
+  };
+
+  changeQuery = (event: { target: { value: string; }; }) => this.setState({
+    query: event.target.value,
+  });
+
+  getFilteredMovies = () => {
+    const { query } = this.state;
+    const queryLow = query.toLocaleLowerCase();
+
+    return [...moviesFromServer].filter(movie => (
```

Comment:
```suggestion
    return moviesFromServer.filter(movie => (
```
filter creates a new array by default